common_benchmark_suite/openxla/benchmark/models/jax/generate_model_artifacts.py

```diff
@@ -0,0 +1,181 @@
+# Copyright 2023 The OpenXLA Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import argparse
+import importlib
+import jax
+import jax.numpy as jnp
+import os
+import pathlib
+import re
+import multiprocessing
+import shutil
+import subprocess
+import sys
+import tarfile
+
+from typing import Any, Optional
+
+# Add openxla dir to the search path.
+sys.path.insert(0, str(pathlib.Path(__file__).parents[4]))
+
+from openxla.benchmark import def_types
+from openxla.benchmark.comparative_suite.jax import model_definitions
+from openxla.benchmark.models.jax import model_interfaces
+
+HLO_FILENAME_REGEX = r".*jit_forward.before_optimizations.txt"
+HLO_STATIC_FILENAME = "xla_hlo_before_optimizations.txt"
+
+
+def _generate_and_save_inputs(model_obj: Any,
+                              model_dir: str) -> tuple[Any, ...]:
+  raw_inputs = model_obj.generate_default_inputs()
+  inputs = model_obj.preprocess(raw_inputs)
+
+  # Save inputs.
+  inputs_dir = os.path.join(model_dir, "inputs")
+  os.makedirs(inputs_dir, exist_ok=True)
+  for idx, input in enumerate(inputs):
+    input_path = os.path.join(inputs_dir, f"input_{idx}.npy")
+    print(f"Saving input {jnp.shape(input)} to {input_path}")
+    jnp.save(input_path, input)
+
+  with tarfile.open(os.path.join(model_dir, "inputs_npy.tgz"), "w:gz") as tar:
+    tar.add(f"{inputs_dir}/", arcname=".")
+  shutil.rmtree(inputs_dir)
+
+  return inputs
+
+
+def _save_outputs(outputs: tuple[Any, ...], model_dir: str):
+  outputs_dir = os.path.join(model_dir, "outputs")
+  os.makedirs(outputs_dir, exist_ok=True)
+
+  for idx, output in enumerate(outputs):
+    output_path = os.path.join(outputs_dir, f"output_{idx}.npy")
+    print(f"Saving output {jnp.shape(outputs)} to {output_path}")
+    jnp.save(output_path, output)
+
+  with tarfile.open(os.path.join(model_dir, "outputs_npy.tgz"), "w:gz") as tar:
+    tar.add(f"{outputs_dir}/", arcname=".")
+  shutil.rmtree(outputs_dir)
+
+
+def _cleanup_hlo(hlo_dir: str, model_dir: str):
```

Comment:
nit: `hlo_dir` and `model_dir` can be `pathlib.Path`

Reply:
Fixed